Sources/Views/MessageInputBar.swift

```diff
@@ -319,6 +319,10 @@ open class MessageInputBar: UIView {
     /// - Parameter positions: The UIStackView's to layout
     public func layoutStackViews(_ positions: [InputStackView.Position] = [.left, .right, .bottom]) {
         
+        guard let _ = superview else {
+            return
```

Comment:
Why are we adding this check ?

Moreover why are we checking this multiple times ? That’s redundant

Reply:
@nathantannar4 , because we should not call `layoutIfNeeded` before `MessageInputBar` be added to the `superview`. If we do this , `Auto Layout` System would use `_UITemporaryLayoutWidth/_UITemporaryLayoutWidth` to break our custom constraints. we can reproduce it when `MessageInputBar` become first responder, and then switch the keyboard .

Log ex.

```
[LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x60400029f630 V:|-(6)-[MessageKit.InputTextView:0x7fe2a20a0200]   (active, names: '|':MessageKit.MessageInputBar:0x7fe2a142b590 )>",
    "<NSLayoutConstraint:0x604000480b40 MessageKit.InputTextView:0x7fe2a20a0200.bottom == MessageKit.InputStackView:0x7fe2a1432e10.top   (active)>",
    "<NSLayoutConstraint:0x604000284fb0 MessageKit.InputStackView:0x7fe2a1432e10.bottom == UILayoutGuide:0x6040007b2c20'UIViewSafeAreaLayoutGuide'.bottom - 6   (active)>",
    "<NSLayoutConstraint:0x60000029e690 '_UITemporaryLayoutHeight' MessageKit.MessageInputBar:0x7fe2a142b590.height == 0   (active)>",
    "<NSLayoutConstraint:0x60400029c110 'UIViewSafeAreaLayoutGuide-bottom' V:[UILayoutGuide:0x6040007b2c20'UIViewSafeAreaLayoutGuide']-(0)-|   (active, names: '|':MessageKit.MessageInputBar:0x7fe2a142b590 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x604000284fb0 MessageKit.InputStackView:0x7fe2a1432e10.bottom == UILayoutGuide:0x6040007b2c20'UIViewSafeAreaLayoutGuide'.bottom - 6   (active)>
```